src/tools/fuzzing/heap-types.cpp

```diff
@@ -736,9 +881,13 @@ void Inhabitator::markNullable(FieldPos field) {
   switch (getVariance(field)) {
     case Covariant:
       // Mark the field null in all supertypes. If the supertype field is
-      // already nullable or does not exist, that's ok and this will have no
-      // effect.
+      // already nullable, that's ok and this will have no effect.
       while (auto super = curr.getDeclaredSuperType()) {
+        if (super->isStruct() && idx >= super->getStruct().fields.size()) {
```

Comment:
Why `>=` - can't this be `==`? I.e., how can idx be higher?

Reply:
If we're coming from a subtype with many more fields, then the index could be arbitrarily far out of bounds when we look at the supertype.

Reply:
I see, thanks.